### PR TITLE
Add LogStreaming for Jobs

### PR DESCRIPTION
**Pull request by lihaMSFT**
**Related command**
`az containerapp job logs show`

**Description**
Add log streaming functionality to jobs

**Testing Guide**
TBD

**History Notes**

[Component Name 1] BREAKING CHANGE: `az command a`: Make some customer-facing breaking change
[Component Name 2] `az command b`: Add some customer-facing feature

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
Thank you for your contribution! We will review the pull request and get back to you soon.

File: src/azure-cli/azure/cli/command_modules/containerapp/_clients.py [hunk 1]
```diff
@@ -16,7 +16,7 @@
 
 logger = get_logger(__name__)
 
-CURRENT_API_VERSION = "2023-05-01"
+CURRENT_API_VERSION = "2023-11-02-preview"
```
Comment (Greedygre) [inline src/azure-cli/azure/cli/command_modules/containerapp/_clients.py:19]: Hi @lihaMSFT 

You cannot change the api-version to 2023-11-02-preview directly. This is a preview feature, please add in azure-cli-extensions like this PR: https://github.com/Azure/azure-cli-extensions/pull/7356

**Comment by zhoxing-ms** [comment]
Please add some test cases for these new feature

**Comment by zhoxing-ms** [comment]
Please resolve these CI issues

**Comment by lihaMSFT** [comment]
This feature is implemented in https://github.com/Azure/azure-cli-extensions/pull/7454

**Pull request closed without merging**